torchhd/datasets/dataset.py

```diff
@@ -7,8 +7,202 @@
+        "Connect4",
+        "Contrac",
+        "CreditApproval",
+        "CylinderBands",
+        "Dermatology",
+        "Echocardiogram",
+        "Ecoli",
+        "EnergyY1",
+        "EnergyY2",
+        "Fertility",
+        "Flags",
+        "Glass",
+        "HabermanSurvival",
+        "HayesRoth",
+        "HeartCleveland",
+        "HeartHungarian",
+        "HeartSwitzerland",
+        "HeartVa",
+        "Hepatitis",
+        "HillValley",
+        "HorseColic",
+        "IlpdIndianLiver",
+        "ImageSegmentation",
+        "Ionosphere",
+        "Iris",
+        "LedDisplay",
+        "Lenses",
+        "Letter",
+        "Libras",
+        "LowResSpect",
+        "LungCancer",
+        "Lymphography",
+        "Magic",
+        "Mammographic",
+        "Miniboone",
+        "MolecBiolPromoter",
+        "MolecBiolSplice",
+        "Monks1",
+        "Monks2",
+        "Monks3",
+        "Mushroom",
+        "Musk1",
+        "Musk2",
+        "Nursery",
+        "OocytesMerlucciusNucleus4d",
+        "OocytesMerlucciusStates2f",
+        "OocytesTrisopterusNucleus2f",
+        "OocytesTrisopterusStates5b",
+        "Optical",
+        "Ozone",
+        "PageBlocks",
+        "Parkinsons",
+        "Pendigits",
+        "Pima",
+        "PittsburgBridgesMaterial",
+        "PittsburgBridgesRelL",
+        "PittsburgBridgesSpan",
+        "PittsburgBridgesTOrD",
+        "PittsburgBridgesType",
+        "Planning",
+        "PlantMargin",
+        "PlantShape",
+        "PlantTexture",
+        "PostOperative",
+        "PrimaryTumor",
+        "Ringnorm",
+        "Seeds",
+        "Semeion",
+        "Soybean",
+        "Spambase",
+        "Spect",
+        "Spectf",
+        "StatlogAustralianCredit",
+        "StatlogGermanCredit",
+        "StatlogHeart",
+        "StatlogImage",
+        "StatlogLandsat",
+        "StatlogShuttle",
+        "StatlogVehicle",
+        "SteelPlates",
+        "SyntheticControl",
+        "Teaching",
+        "Thyroid",
+        "TicTacToe",
+        "Titanic",
+        "Trains",
+        "Twonorm",
+        "VertebralColumn2Clases",
+        "VertebralColumn3Clases",
+        "WallFollowing",
+        "Waveform",
+        "WaveformNoise",
+        "Wine",
+        "WineQualityRed",
+        "WineQualityWhite",
+        "Yeast",
+        "Zoo",
+    ]
+
+    # Specify namedtuple format
+    DATASET = namedtuple("Dataset", ["name", "train", "test"])
+
+    def __init__(
+        self,
+        root: str,
+        download: bool,
+    ):
+        super(UCIClassificationBenchmark, self).__init__()
+        self.root = root
+        self.download = download
+        self.statistics = {key: [0.0, 0.0] for key in self.UCI_DATASET_COLLECTION}
+
+    # DK there could be better a way to make this generator within the class but I did not find it
+    def datasets(self):
+        # For all datasets in the collection
+        for i in range(len(self.UCI_DATASET_COLLECTION)):
```

Comment:
Since you don't use `i` besides getting the dataset name I would instead opt to do `for dataset_name in self.UCI_DATASET_COLLECTION`. But that's just me nitpicking.